src/account/UpgradeableModularAccount.sol

```diff
@@ -310,6 +318,25 @@ contract UpgradeableModularAccount is
         super.upgradeToAndCall(newImplementation, data);
     }
 
+    function isValidSignature(bytes32 hash, bytes calldata signature) public view override returns (bytes4) {
+        AccountStorage storage _storage = getAccountStorage();
+
+        FunctionReference sigValidation = FunctionReference.wrap(bytes21(signature));
+
+        (address plugin, uint8 functionId) = sigValidation.unpack();
+        if (!_storage.signatureValidations.contains(toSetValue(sigValidation))) {
+            revert SignatureValidationInvalid(plugin, functionId);
+        }
+
+        if (
+            IValidation(plugin).validateSignature(functionId, msg.sender, hash, signature[21:])
+                == _1271_MAGIC_VALUE
```

Comment:
Since `IValidation.validateSignature` returns `MAGIC_VALUE` or `INVALID`, we could just do `return (IValidation(plugin).validateSignature(functionId, msg.sender, hash, signature[21:])` right?